### ISSUE TITLE
New instance of the tour created on every render of the provider

### ISSUE THREAD

**Issue by nathggns** [shipshapecode/react-shepherd#282]
As the tour instance is created in the render block of the context provider component, every time this is re-rendered this creates a new instance of the tour, which is then provided to child components via the context. 

If you're trying to trigger the tour automatically using an effect, this will result in the tour being re-created every time the provider component is re-rendered.

A potential fix for this is moving the instantiation of the tour into an effect in the provider component, but I haven't investigated this yet. 

Check out this sandbox for a reproduction of the bug. https://codesandbox.io/s/optimistic-sea-96fby

**Comment by nathggns** [comment]
Actually probably don't need an effect. A useMemo is probably fine for it.

**Comment by chuckcarpenter** [comment]
@nathggns I should've tested this in a React app, but doing this now. Seeing this error:

```
Error: Invalid hook call. Hooks can only be called inside of the body of a function component. This could happen for one of the following reasons:
1. You might have mismatching versions of React and the renderer (such as React DOM)
2. You might be breaking the Rules of Hooks
3. You might have more than one copy of React in the same app
See https://fb.me/react-invalid-hook-call for tips about how to debug and fix this problem.
```

**Comment by chuckcarpenter** [comment]
okay, fixed in #285 if you want to check it out. Specifically, redundant React versions: https://github.com/shipshapecode/react-shepherd/pull/285/commits/dbebe72109269394a3d4d88b967f154c6c2d6543#diff-b9cfc7f2cdf78a7f4b91a753d10865a2L71
